src/Main.java

```diff
@@ -0,0 +1,173 @@
+// 1. Maak een main-klasse aan in de src-map.
+public class Main {
+// 2. Maak vervolgens binnen de main-klasse een main-methode aan. Anders kun je de applicatie immers niet draaien!
+    public static void main(String[] args) {
+// 5. Instantieer in de main methode jouw 4 verschillende Pokèmon met de verschillende attributen.
+        ElectricPokemon pokemon1 = new ElectricPokemon("Pikachu", 10, 100, 0);
+        FirePokemon pokemon2 = new FirePokemon("Charizard", 50, 200, 0, 80);
+        WaterPokemon pokemon3 = new WaterPokemon("Blastoise", 60, 250, 0, 90);
+        GrassPokemon pokemon4 = new GrassPokemon("Venusaur", 55, 220, 0, 85);
+
+// 6. Voer in de main methode verschillende methodes uit van verschillende Pokemons, uit zowel de subklassen als uit de super klasse.
+        pokemon1.thunderPunch();
+        pokemon2.inferno();
+        pokemon3.hydroPump();
+        pokemon4.solarBeam();
+    }
+}
+// 3. Maak een Pokèmon Super klasse aan en zorg ervoor dat deze voldoet aan de gestelde randvoorwaarden
+class Pokemon {
+    private String name;
+    private int level;
+    private int hp;
+    private int xp;
+
+    public Pokemon(String name, int level, int hp, int xp) {
+        this.name = name;
+        this.level = level;
+        this.hp = hp;
+        this.xp = xp;
+    }
+
+    // Override
+    public void speak() {
+        System.out.println(name + " is speaking.");
+    }
+
+    // Getters and setters
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+    public void setLevel(int level) {
+        this.level = level;
+    }
+
+    public int getHp() {
+        return hp;
+    }
+    public void setHp(int hp) {
+        this.hp = hp;
+    }
+
+    public int getXp() {
+        return xp;
+    }
+    public void setXp(int xp) {
+        this.xp = xp;
+    }
+}
+
+
+// 4. Maak de volgende subklassen aan en laat deze voldoen aan de gestelde randvoorwaarden.
+// Subklasse: ElectricPokemon
+class ElectricPokemon extends Pokemon {
+    private int electricPower;
+    public ElectricPokemon(String name, int level, int hp, int xp) {
+        super(name, level, hp, xp);
+        this.electricPower = electricPower;
+    }
+    public void thunderPunch() {
+        System.out.println(getName() + " uses Thunder Punch!");
```

Comment:
`getName()` werkt goed, maar probeer het de volgende keer te schrijven als `super.getName()` daarmee is het direct duidelijk dat dit een methode van de superklasse is.